src/h5web/providers/mock/MockProvider.tsx

```diff
@@ -1,22 +1,38 @@
 import type { ReactElement, ReactNode } from 'react';
+import { assertDefined, assertGroup } from '../../guards';
+import { getEntityAtPath } from '../../utils';
 import Provider from '../Provider';
 import { mockMetadata, mockValues, mockDomain } from './data';
 
 interface Props {
   domain?: string;
+  slowOnPath?: string;
   errorOnId?: string;
   children: ReactNode;
 }
 
 function MockProvider(props: Props): ReactElement {
-  const { domain = mockDomain, errorOnId, children } = props;
+  const { domain = mockDomain, errorOnId, slowOnPath, children } = props;
 
   return (
     <Provider
       api={{
         domain,
-        fetchMetadata: async () => mockMetadata,
-        fetchValue: async (id: keyof typeof mockValues) => {
+        getMetadata: async () => mockMetadata,
+        getGroup: async (path: string) => {
+          if (path === slowOnPath) {
+            await new Promise((resolve) => {
+              setTimeout(resolve, 3000);
+            });
+          }
+
+          const group = getEntityAtPath(mockMetadata, path, true);
```

Comment:
Note that we use `getEntityAtPath` here. We could go back to a flat structure for the mock metadata (with groups stored against their paths), but the nested structure is a lot more readable...